_release/bat/storage_bat/storage_bat_test.go

```diff
@@ -735,3 +735,90 @@ func TestCreateVolumeFromVolume(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// Check that volumes created from image with a size override work
+//
+// Create a volume from an image with a size and check new size allocated
+//
+// The created volume should be the newly requested size
+func TestCreateSizedVolumeFromImage(t *testing.T) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), standardTimeout)
+	defer cancelFunc()
+
+	options := bat.ImageOptions{
+		Name: "test-image",
+	}
+
+	image, err := bat.AddRandomImage(ctx, "", 10, &options)
+	if err != nil {
+		t.Fatalf("Unable to add image %v", err)
+	}
+
+	volumeUUID, err := bat.AddVolume(ctx, "", image.ID, "image", &bat.VolumeOptions{Size: 10})
```

Comment:
Unit difference, the image size is specified is 10MiB, but i'm resizing to 10GiB.